### ISSUE TITLE
Have server get answers from upstream DNS server?

### ISSUE THREAD

**Issue by phpdave11** [bluejekyll/trust-dns#436]
I am trying to switch my BIND9 setup to trust-dns server.  I have everything working with my local domains, but when I try to resolve a public domain, I get no answers.

On the server that is running trust-dns server, I have upstream DNS servers defined in `/etc/resolv.conf`:

```
nameserver 208.67.222.222
nameserver 208.67.220.220
```

Here's what I get when I try to query trust-dns for a public domain:
```
$ dig @10.42.0.25 github.com

; <<>> DiG 9.12.1 <<>> @10.42.0.25 github.com
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 5828
;; flags: qr; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags: do; udp: 4096
; OPT=6: 08 0d 0e 0f ("....")
; OPT=5: 08 0d 0e 0f ("....")
;; QUESTION SECTION:
;github.com.                    IN      A

;; Query time: 0 msec
;; SERVER: 10.42.0.25#53(10.42.0.25)
;; WHEN: Sat Apr 28 10:58:00 CDT 2018
;; MSG SIZE  rcvd: 55
```

I started with the example named configuration files.  Am I missing some configuration, or is this not supported?

Thanks!

**Comment by bluejekyll** [comment]
You are missing something, and it's a very important feature we still need to build! Please see #55

Forwarding support is not yet complete. With the Resolver (limited to stub resolution at-the-moment) work in a pretty decent state at this point, I do see us in a position now to integrate the Resolver into the server for forwarding support. Right now, the trust-dns server is only capably of acting as an authority.

I'm gearing up to work on the resolver integration soon, but I don't anticipate that work being complete for a while. It's going to require some refactoring of the Resolver and the Server before such a thing will be ready.

I hope this wasn't wasted effort for you!

**Comment by phpdave11** [comment]
No problem, thanks for the explanation -- can't wait for forwarding support!

Nice job on this whole project by the way!  It was definitely not wasted effort on my part, as I too am trying to learn rust and get more familiar with programs written in rust.

**Comment by bluejekyll** [comment]
fixed in #674
